esmvaltool/diag_scripts/hydrology/hydro_forcing.py

```diff
@@ -0,0 +1,203 @@
+
+    # Store provenance
+    log_provenance(caption, filename_plot, cfg, ancestors)
+
+
+def plot_timeseries_data(cfg,
+                         metadata,
+                         time_unit: str = 'D',
+                         title: str = 'plot'):
+    """Plot timeseries data."""
+    datasets = read_input_data(metadata)
+    ancestors = [info['filename'] for info in metadata]
+
+    var = datasets.pr
+
+    start_date = np.datetime_as_string(datasets.time.min(), unit=time_unit)
+    end_date = np.datetime_as_string(datasets.time.max(), unit=time_unit)
+
+    name = title.lower().replace(' ', '_')
+    caption = f"{title} for {start_date}:{end_date}"
+
+    plot_data(
+        cfg=cfg,
+        datasets=datasets,
+        xaxis='time',
+        yaxis='pr',
+        xlabel='time / {mip}'.format(**datasets.attrs),
+        ylabel=f'{var.long_name} / {var.units}',
+        caption=caption,
+        name=name,
+        ancestors=ancestors,
+    )
+
+    filename_data = get_diagnostic_filename(name, cfg, extension='nc')
+    datasets.to_netcdf(filename_data)
+    log_provenance(caption, filename_data, cfg, ancestors)
+
+
+def plot_climatology(cfg, metadata, title: str = 'plot'):
+    """Plot climatology data."""
+    datasets = read_input_data(metadata)
+    var = datasets.pr
+
+    ancestors = [info['filename'] for info in metadata]
+
+    name = title.lower().replace(' ', '_')
+    caption = f"{title}"
+
+    plot_data(
+        cfg=cfg,
+        datasets=datasets,
+        xaxis='month_number',
+        yaxis='pr',
+        xlabel='Month number',
+        ylabel=f'{var.long_name} / {var.units}',
+        caption=caption,
+        name=name,
+        ancestors=ancestors,
+    )
+
+    filename_data = get_diagnostic_filename(name, cfg, extension='nc')
+    datasets.to_netcdf(filename_data)
+    log_provenance(caption, filename_data, cfg, ancestors)
+
+
+def read_input_data(metadata: list, dim: str = 'dataset'):
+    """Load data from metadata.
+
+    Read the input data from the list of given data sets. `metadata` is
+    a list of metadata containing the filenames to load. The datasets
+    are stacked along the `dim` dimension. Returns an xarray.DataArray.
+    """
+    identifiers = []
+    datasets = []
+    for info in metadata:
+        filename = info['filename']
+        dataset = xr.load_dataset(filename)
+        datasets.append(dataset)
+
+        identifier = info[dim]
+        identifiers.append(identifier)
+
+    stacked_datasets = xr.concat(datasets, dim=dim)
+    stacked_datasets[dim] = identifiers
+
+    return stacked_datasets
+
+
+def get_diagnostic_data(cfg):
+    """Get diagnostic name and data."""
+    input_data = cfg['input_data'].values()
+    diagnostic_data = group_metadata(input_data, 'diagnostic')
+    return list(diagnostic_data.items())[0]
+
+
+def main(cfg):
+    """Load and plot hydro forcing data."""
+    diagnostic, metadata = get_diagnostic_data(cfg)
+
+    if diagnostic == 'sample_year':
```

Comment:
I really don't like using the name of the diagnostic to define its behaviour. It's obscure and can mislead the users. It also imposes some limitations on the recipe itself, like being unable to do the detailed year plot for more than one year in a single run.

Can you please change it to a user setting and update the doc?

Reply:
Thanks, I was not aware of this convention. I had this at first, but it seemed a bit redundant to me to use a setting to pass the name of the diagnostic while it is already available, which is why I looked for an alternative. 

Will change it back 👍

Reply:
Thanks @stefsmeets, yes this looks much better.

I noticed you use "entry_point" as the name of the setting. I do not think this is very clear to the user (entry into what?). 

Perhaps change it to "plot_type" as also used in the [Python example](https://github.com/ESMValGroup/ESMValTool/blob/master/esmvaltool/recipes/examples/recipe_python.yml)?

Reply:
Sure, I have updated it to `plot_type` 👍